src/PrestaShopBundle/Bridge/Helper/Listing/HelperBridge/HelperListBridge.php

```diff
@@ -102,33 +103,30 @@ public function __construct(
      * Generate the html for list using HelperList class
      *
      * @param HelperListConfiguration $helperListConfiguration
+     * @param Request $request
      *
      * @return string|null
      */
     public function generateList(
-        HelperListConfiguration $helperListConfiguration
+        HelperListConfiguration $helperListConfiguration,
+        Request $request
     ): ?string {
         if (!($helperListConfiguration->fieldsList && is_array($helperListConfiguration->fieldsList))) {
             return null;
         }
 
-        $this->generateListQuery($helperListConfiguration, $this->context->language->id);
+        $this->generateListQuery($helperListConfiguration, $request, $this->context->language->id);
 
         $helper = new HelperList();
 
         $this->helperListConfigurator->setHelperDisplay($helperListConfiguration, $helper);
-        $helper->_default_pagination = $helperListConfiguration->defaultPagination;
-        $helper->_pagination = $helperListConfiguration->pagination;
-        $helper->tpl_delete_link_vars = $helperListConfiguration->deleteLinksVariableTemplate;
-
-        foreach ($helperListConfiguration->actionsAvailable as $action) {
-            if (!in_array($action, $helperListConfiguration->actions) && isset($helperListConfiguration->$action) && $helperListConfiguration->$action) {
-                $helperListConfiguration->actions[] = $action;
-            }
-        }
+        $helper->_default_pagination = $helperListConfiguration->getDefaultPaginationLimit();
+        $helper->_pagination = $helperListConfiguration->getPaginationLimits();
+        $helper->tpl_delete_link_vars = $helperListConfiguration->getDeleteLinkVars();
+        $helper->filtersSubmitUrl = $helperListConfiguration->getFiltersSubmitUrl();
 
         /* @phpstan-ignore-next-line */
-        $helper->sql = $helperListConfiguration->listsql;
+        $helper->sql = $helperListConfiguration->listSql;
```

Comment:
Seems like `$helperListConfiguration->listSql` is only used as a temporary container while building the query in `generateListQuery` It's the only place where it's accessed outside of the method.

So it seems like instead of storing this into the `$helperListConfiguration`, it should be generated as a local variable and returned by the `generateListQuery` method